src/libsync/propagatedownload.cpp

```diff
@@ -378,6 +386,74 @@ QString GETFileJob::errorString() const
     return AbstractNetworkJob::errorString();
 }
 
+GETEncryptedFileJob::GETEncryptedFileJob(AccountPtr account, const QString &path, QIODevice *device,
+    const QMap<QByteArray, QByteArray> &headers, const QByteArray &expectedEtagForResume,
+    qint64 resumeStart, EncryptedFile encryptedInfo, QObject *parent)
+    : GETFileJob(account, path, device, headers, expectedEtagForResume, resumeStart, parent)
+    , _encryptedFileInfo(encryptedInfo)
+{
+}
+
+GETEncryptedFileJob::GETEncryptedFileJob(AccountPtr account, const QUrl &url, QIODevice *device,
+    const QMap<QByteArray, QByteArray> &headers, const QByteArray &expectedEtagForResume,
+    qint64 resumeStart, EncryptedFile encryptedInfo, QObject *parent)
+    : GETFileJob(account, url, device, headers, expectedEtagForResume, resumeStart, parent)
+    , _encryptedFileInfo(encryptedInfo)
+{
+}
+
+qint64 GETEncryptedFileJob::writeToDevice(const char *data, qint64 len)
```

Comment:
and here the return value is always the full length or -1 to indicate an error smells like something wrong in the API

Reply:
@mgallien Not sure I follow. What are you proposing to change?

Reply:
@mgallien Well, I've been trying to make this `writeToDevice` compatible with `write` which returns an int. If I use `Optional` or, whatever, I would then need to change logic inside `void GETFileJob::slotReadyRead()` and will not be able to log `bytes read` and `bytes written`. What is the main concern of using int and `-1`? `QIODevice::write` is using this convention already with no issues whatsoever, so, I am in doubt why we don't want to adapt it since we are using Qt?